content/setup/local-with-vagrant/_index.md

```diff
@@ -0,0 +1,198 @@
+
+```
+$ cat > hello-world.yml
+version: "0.1"
+name: hello_world_workflow
+global_timeout: 600
+tasks:
+  - name: "hello world"
+    worker: "{{.device_1}}"
+    actions:
+      - name: "hello_world"
+        image: hello-world
+        timeout: 60
+$ docker exec -i deploy_tink-cli_1 tink template create --name hello-world < ./hello-world.yml
+Created Template:  75ab8483-6f42-42a9-a80d-a9f6196130df
+```
+
+Register the worker with Tinkerbell
+
+```
+$ cat > hardware-data.json
+{
+  "id": "ce2e62ed-826f-4485-a39f-a82bb74338e2",
+  "arch": "x86_64",
+  "allow_pxe": true,
+  "allow_workflow": true,
+  "facility_code": "onprem",
+  "ip_addresses": [
+    {
+      "address": "192.168.1.5",
+      "address_family": 4,
+      "enabled": true,
+      "gateway": "192.168.1.1",
+      "management": true,
+      "netmask": "255.255.255.248",
+      "public": false
+    }
+  ],
+  "network_ports": [
+    {
+      "data": {
+        "mac": "08:00:27:00:00:01"
+      },
+      "name": "eth0",
+      "type": "data"
+    }
+  ]
+}
+
+$ docker exec -i deploy_tink-cli_1 tink hardware push < ./hardware-data.json
+2020/06/17 14:12:45 Hardware data pushed successfully
+```
+
+Now that we have a template and hardware registered we can trigger the workflow for that particular hardware but first a bit of context on what we just did:
+
+1. A template is used to create workflows. Template are important because you
+   can create as many workflows you need from a single template.
+2. A workflow needs a target (a hardware) in vagrant it is the worker and it is
+   identified via MAC address.
+
+Let's start the workflow where `-t` is the template ID you got when registering
+your template and `-r` contains the variables to replace in the template, in
+this case only the worker's MAC address is required:
+
+```
+$ docker exec -i deploy_tink-cli_1 tink workflow create \
+    -t 75ab8483-6f42-42a9-a80d-a9f6196130df \
+    -r '{"device_1":"08:00:27:00:00:01"}'
+Created Workflow:  a8984b09-566d-47ba-b6c5-fbe482d8ad7f
+```
+
+> Note: this MAC address it hard coded in the Vagrantfile.
+
+We need to spin up our first worker now.
+
+## Start a worker
+
+Open a new terminal and get inside the `tink/deploy/vagrant`.
+
+For a worker the MAC address works as a unique identifier, with Vagrant we are
+able to set it to fixed value, in this case `080027000001`.
+
+As you did for the provisioner start the worker via vagrant:
+
+```
+$ vagrant up worker
+```
+
+If using VirtualBox, the worker shows up in a UI and if everything is right you
+should see at the end of the boot a login to a custom Alpine that Tinkerbell
+netboot. The operating system runs on RAM, the changes won't be persisted across
+reboots. You can login with the username `root` and no password is required.
+
+![Screenshot from the worker](/images/vagrant-setup-vbox-worker.png)
+
+That's it! Let us know what you think about it on [Slack](/community-slack).
```

Comment:
For me this finishes too early. Yes I brought up the worker, but it gave no output about being provisioned

My suggestion for completing the story:

Run `tink-cli` with `workflow` and with `events` to show that hello-world actually executed.

Mention also that they have to create a new workflow for each boot of the machine. 

Also show them how to get logs, this is also missing but a key piece of using Tinkerbell.

I assume that would be the following - with a brief explanation on the debug data that you can find in each one.

```
docker-compose logs tink-server
docker-compose logs boots
docker-compose logs nginx
```

Reply:
For example:

<img width="2542" alt="Screenshot 2020-06-18 at 12 20 36" src="https://user-images.githubusercontent.com/6358735/85014409-212f1f80-b15e-11ea-8643-fd940c4d919a.png">

And you can see that the workflow (following the tutorial) didn't actually work / execute since there are no events. 

<img width="2346" alt="Screenshot 2020-06-18 at 12 21 34" src="https://user-images.githubusercontent.com/6358735/85014506-4e7bcd80-b15e-11ea-827e-e54ac57e6d38.png">

Reply:
Hello! Yeah, I am sure there are a lot more we can explain. We are not probably the unique two with opinions here 💯 I presume we will figure out what to add to this documentation next week.

Thanks for your screenshots and tips